src/main/java/rx/Single.java

```diff
@@ -1569,14 +1569,12 @@ public final void onNext(T args) {
      * @param subscriber
      *            the Subscriber that will handle the emission or notification from the Single
      */
-    public final void unsafeSubscribe(Subscriber<? super T> subscriber) {
+    public final Subscription unsafeSubscribe(Subscriber<? super T> subscriber) {
```

Comment:
Single isn't stable API so this is acceptable if it makes the API better.

On Fri, Feb 19, 2016, 10:46 AM Artem Zinnatullin <notifications@github.com>
wrote:

> In src/main/java/rx/Single.java
> <https://github.com/ReactiveX/RxJava/pull/3696#discussion_r53476625>:
>
> > @@ -1569,14 +1569,12 @@ public final void onNext(T args) {
> >       * @param subscriber
> >       *            the Subscriber that will handle the emission or notification from the Single
> >       */
> > -    public final void unsafeSubscribe(Subscriber<? super T> subscriber) {
> > +    public final Subscription unsafeSubscribe(Subscriber<? super T> subscriber) {
>
> Looks like this is incompatible binary change
> https://docs.oracle.com/javase/specs/jls/se7/html/jls-13.html#jls-13.4.15
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ReactiveX/RxJava/pull/3696/files#r53476625>.
>